x/marker/types/key_test.go

```diff
@@ -34,3 +34,13 @@ func TestDenySendKey(t *testing.T) {
 	assert.Equal(t, denyAddr.Bytes(), denyKey[denomArrLen+3:denomArrLen+3+denyAddrLen], "should match deny key")
 	assert.Len(t, denyKey, int(3+denomArrLen+denyAddrLen), "should have key of length of sum 1 for prefix 2 length bytes and length of denom and deny address")
 }
+
+func TestGetDenySendAddresses(t *testing.T) {
+	addr, err := MarkerAddress("nhash")
+	require.NoError(t, err)
+	denyAddr := sdk.AccAddress("cosmos1v57fx2l2rt6ehujuu99u2fw05779m5e2ux4z2h")
+	denyKey := DenySendKey(addr, denyAddr)
+	mAddr, dAddr := GetDenySendAddresses(denyKey)
+	assert.Equal(t, mAddr, addr)
+	assert.Equal(t, dAddr, denyAddr)
```

Comment:
Your expected and actual are backwards in this call: `Equal(t TestingT, expected interface{}, actual interface{}, msgAndArgs ...interface{})`. Additionally, **ALWAYS** provide a message with your assertions. That way, when it fails, the failure message tells you what you're looking at.

```suggestion
	assert.Equal(t, addr, mAddr, "module address")
	assert.Equal(t, denyAddr, dAddr, "deny address")
```

Reply:
Oops, mistake on the order. I will add the messages, but I don't think that is always necessary. The test name, line number variable names, and expected values messages are usually enough.

Reply:
Instances where a msg/args isn't necessary are very rare. The line number only helps you when you're looking at the test code and when no code has changed yet in the file (to alter the line number). When you rely on the line number, you have to do this back and forth comparing the failure output and the code. That becomes even more painful when the error message isn't available in your IDE (e.g. it's in a Github Action).

When you include a message, you're putting that information in the test failure message with all the other information about the failure; it's all in one place. It also makes it easier to identify exactly what is wrong which often lets you skip digging into the unit test and instead go directly to the production code to fix the problem.

In this case, you've got two `Equals` tests on byte slices. The values in the failure message are not going to provide any clue as to which byte slices you're looking at. Knowing which one isn't as expected tells you exactly what to look at in `GetDenySendAddresses`.

In the rare cases where a msg/args might not be necessary, having them is often still helpful, and are only detrimental if what they contain isn't accurate. Once you're used to always adding them, they take almost no time to add. And having them will save you plenty of time in the end.

Always including them in your assertions is a good habit to get into.

Reply:
Also, the `.NoError` above should have a message too since the error isn't coming from the `GetDenySendAddresses` that the test is about. As it is, if there's an error there, you'll probably waste time looking at `GetDenySendAddresses` first, only to find it doesn't return an error. Then, you'd go to the test code to look up what exactly was called that returned the error.

If you include the msg "MarkerAddress(nhash)" in that assertion, when it fails, you'll know exactly what's broken, and that that unit test will (probably) be fixed when you fix `MarkerAddress`. You don't even have to look at the unit test code. You just go to `MarkerAddress`, identify and fix the problem, and rerun unit tests (since lots of things probably failed as well).